fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBRecordStoreQueryTestBase.java

```diff
@@ -113,25 +116,44 @@ protected void setupSimpleRecordStore(RecordMetaDataHook recordMetaDataHook,
 
     protected int querySimpleRecordStore(RecordMetaDataHook recordMetaDataHook, RecordQueryPlan plan,
                                          Supplier<EvaluationContext> contextSupplier,
-                                         TestHelpers.DangerousConsumer<TestRecords1Proto.MySimpleRecord.Builder> checkRecord)
+                                         TestHelpers.DangerousConsumer<TestRecords1Proto.MySimpleRecord> checkRecord)
             throws Exception {
         return querySimpleRecordStore(recordMetaDataHook, plan, contextSupplier, checkRecord, context -> { });
     }
 
     protected int querySimpleRecordStore(RecordMetaDataHook recordMetaDataHook, RecordQueryPlan plan,
                                          Supplier<EvaluationContext> contextSupplier,
-                                         TestHelpers.DangerousConsumer<TestRecords1Proto.MySimpleRecord.Builder> checkRecord,
+                                         TestHelpers.DangerousConsumer<TestRecords1Proto.MySimpleRecord> checkRecord,
```

Comment:
This change isn't strictly necessary for this PR. The main change is to use the immutable `MySimpleRecord` class for assertions instead of the mutable `MySimpleRecord.Builder`s. None of the assertions were mutating the records, but it felt strange that the API could support it, so I modified it to remove it. It does mean that we need to call `.build()` before calling the assertions, but that should be a relatively minor thing (to be clear: in test code)